app/assets/javascripts/controllers/network_router/network_router_form_controller.js

```diff
@@ -1,33 +1,70 @@
-ManageIQ.angular.app.controller('networkRouterFormController', ['$http', '$scope', 'networkRouterFormId', 'miqService', function($http, $scope, networkRouterFormId, miqService) {
+ManageIQ.angular.app.controller('networkRouterFormController', ['networkRouterFormId', 'miqService', 'API', function(networkRouterFormId, miqService, API) {
   var vm = this;
 
+  vm.afterGet = false;
   vm.networkRouterModel = {
     name: '',
-    cloud_subnet_id: '',
+    cloud_subnet_id: null,
+    cloud_tenant: null,
+    enable_snat: true,
+    external_gateway: false,
+    extra_attributes: null,
   };
+
   vm.formId = networkRouterFormId;
-  vm.afterGet = false;
-  vm.modelCopy = angular.copy( $scope.networkRouterModel );
   vm.model = "networkRouterModel";
-
+  vm.newRecord = networkRouterFormId === "new";
   vm.saveable = miqService.saveable;
 
-  ManageIQ.angular.scope = vm;
-
-  if (networkRouterFormId == 'new') {
-    vm.networkRouterModel.name = "";
-    vm.networkRouterModel.enable_snat = true;
-    vm.networkRouterModel.external_gateway = false;
-    vm.networkRouterModel.cloud_subnet_id = null;
-    vm.newRecord = true;
+  if (vm.newRecord) {
+    vm.afterGet = true;
+    vm.modelCopy = angular.copy(vm.networkRouterModel);
   } else {
     miqService.sparkleOn();
-
-    $http.get('/network_router/network_router_form_fields/' + networkRouterFormId)
-      .then(getNetworkRouterFormData)
-      .catch(miqService.handleFailure);
+    API.get("/api/network_routers/" + networkRouterFormId + "?attributes=name,ems_id,admin_state_up,cloud_network_id,extra_attributes,cloud_tenant,ext_management_system,cloud_subnets").then(function(data) {
+      Object.assign(vm.networkRouterModel, data);
+      if (data.extra_attributes.external_gateway_info && data.extra_attributes.external_gateway_info !== {}) {
+        vm.networkRouterModel.external_gateway = true;
+      }
+    }).then(function() {
+      if (vm.networkRouterModel.external_gateway) {
+        getSubnetByRef(vm.networkRouterModel.extra_attributes.external_gateway_info.external_fixed_ips[0].subnet_id);
```

Comment:
`return getSubnetByRef(...` so it waits on `Promise` that you should return from this function. If you want one `.then` to be finished before next one starts.